checkout/access-checkout-android/src/main/java/com/worldpay/access/checkout/views/AbstractCardFieldLengthFilter.kt

```diff
@@ -40,17 +40,17 @@ sealed class AbstractCardFieldLengthFilter(private val cardConfiguration: CardCo
     }
 
     private fun getMaxForDefaultRules(): Int? {
-        return ruleSelectorForDefaults(cardConfiguration?.defaults)?.let { getValueToUseForRule(it) }
+        return ruleSelectorForDefaults(cardConfiguration?.defaults)?.let { getValueToUseForRule(it)?.max() }
     }
 
     private fun getMaxForIdentifiedBrand(result: Pair<ValidationResult, CardBrand?>, spanned: Spanned): Int? {
         return result.second?.let {
-            ruleSelectorForCardBrand(it, spanned)?.let { rule -> getValueToUseForRule(rule) }
+            ruleSelectorForCardBrand(it, spanned)?.let { rule -> getValueToUseForRule(rule)?.max() }
         }
     }
 
-    private fun getValueToUseForRule(cardValidationRule: CardValidationRule): Int? {
-        return cardValidationRule.maxLength ?: cardValidationRule.validLength ?: cardValidationRule.minLength
+    private fun getValueToUseForRule(cardValidationRule: CardValidationRule): List<Int>? {
+        return cardValidationRule.validLengths
```

Comment:
change this to return the max instead, and rename the function

Reply:
updated